### PR TITLE
fix/authentication/signup-ui-tests : Fix UI tests for M1: SignUp

### PR DESCRIPTION
**Pull request by taghizadlaura**
# Fix UI tests for M1: SignUp

## Description
This PR introduces UI tests for the SignUpScreen component in the authentication module. It validates various user input scenarios to ensure proper error handling and navigation flow. It closes issue #76.

## Changes
I added more tests, including tests for the calls to the ViewModel. Now the navigation tests also work.

#### Modified
- `SignUpScreenTest.kt`

### CODE REVIEW

**Review by charliemangano — CHANGES_REQUESTED** [review]
## Summary
These tests are very complete, good job! Small changes here and there, check the comments!
## Important
#### Code Quality
Very well formatted code, very consistent as well. I like that you extracted email and password variations in a companion object, it helps with consistency and error checking.
#### Functionality
I think the tests check for the proper things, and correctly. Be careful of edge cases though, see comments.
## Testing
The tests pass and are very  complete! Great job:)
## Documentation
Code is self-documenting thanks to the function and variable clear nomenclature!
## Suggestions
See comments
## Steps before PR approved
See comments. Any comment that is not annotated with "nitpick" has to be resolved before I approve this PR.

File: app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt [hunk 1]
```diff
@@ -1,172 +1,303 @@
...
+  @Test
+  fun validSignUpAttemptNavigatesToCreateProfileScreen() {
+
+    composeTestRule.onNodeWithTag("signUpEmail").performTextInput(email)
+    composeTestRule.onNodeWithTag("signUpPassword").performTextInput(password)
+    composeTestRule.onNodeWithTag("signUpConfirmPassword").performTextInput(password)
+    composeTestRule.onNodeWithTag("signUpButton").performClick()
+    verify(navigationActions).navigateTo(Screen.CREATE_PROFILE)
+
+    // TODO: Supabase integration for account creation
```
[the remark this answers is not in the document]
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt:289]: I had put this at the very begging when implementing the UI. That was were I though the account validation with Supabase would be done, but it didn't in the end. So, I think delete that line 289.

File: app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt [hunk 2]
```diff
@@ -1,172 +1,303 @@
...
+  }
+
+  @Before
+  fun setUp() {
+    navigationActions = mock(NavigationActions::class.java)
+    authViewModel = mock(AuthViewModel::class.java)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Route.ALERT_LIST)
+    `when`(authViewModel.userAuthState)
+        .thenReturn(mutableStateOf(UserAuthState.Success("User is logged up")))
```
[the remark this answers is not in the document]
Comment (taghizadlaura) [reply app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt:51]: wouldn't it be "signed up" ?

[the remark this answers is not in the document]
Comment (charliemangano) [reply app/src/androidTest/java/com/android/periodpals/ui/authentication/SignUpTest.kt]: ah yes, you're right

**Review by francelu — APPROVED** [review]
## Summary
LGTM. 

## Important
#### Code Quality
- The separation of tests for "doesNotCallVM" seems redundant but maintains consistency with `SignInScreen`.

#### Functionality
- The tests correctly validate expected behaviors.

## Steps before PR approved
- None, keep up the good work!

**Review by charliemangano — APPROVED** [review]
Thanks for applying the changes! All tests pass and seem correct:)
LGTM
